cpp/core_test.cc

```diff
@@ -23,6 +28,109 @@ TEST exceptions_test() {
   PASS();
 }
 
+#if RET_VAL_ROOTING
+TEST environ_test() {
+  Dict<Str*, Str*>* env = pyos::Environ();
+  Str* p = env->get(StrFromC("PATH"));
+  ASSERT(p != nullptr);
+  log("PATH = %s", p->data_);
+
+  PASS();
+}
+
+TEST user_home_dir_test() {
+  uid_t uid = getuid();
+  Str* username = pyos::GetUserName(uid);
+  ASSERT(username != nullptr);
+
+  Str* dir0 = pyos::GetMyHomeDir();
+  ASSERT(dir0 != nullptr);
+
+  Str* dir1 = pyos::GetHomeDir(username);
+  ASSERT(dir1 != nullptr);
+
+  ASSERT(str_equals(dir0, dir1));
+
+  PASS();
+}
+
+TEST uname_test() {
+  Str* os_type = pyos::OsType();
+  ASSERT(os_type != nullptr);
+
+  utsname un = {};
+  ASSERT(uname(&un) == 0);
+  ASSERT(str_equals(StrFromC(un.sysname), os_type));
+
+  PASS();
+}
+
+TEST pyos_readbyte_test() {
```

Comment:
These are from leaky_core_test.cc right?

I think we should remove those

Or is it hard to make them run in both modes?

Reply:
Yeah, there's still some work left to get this working with `StackRoots`. Simply adding it wherever I added `RootingScope` wasn't enough

Reply:
OK I think we may have an unresolved problem here

I'm not sure if it's worth making them run in both modes, but I also don't want to lose test coverage for too long

We might just need to "tear the band-aid off" and go for it, but we should have all the ducks in a row

Reply:
Now that your change is in I moved everything from `leaky_core_test` into here and made the relevant bindings unleaky.